### PR TITLE
filters are added for products

### PR DESCRIPTION
**Pull request by parulgupta2001**

### CODE REVIEW

**Comment by srinibas39** [comment]
Hey parul , every things looks good but i could suggest you one thing,
When your filtering categories instead of sending payload for each category, you can think of a logic where you to have send a single payload for filtering categories.So tommorow if your no of category increases in your backend you dont have to write separate code to handle it.

**Comment by parulgupta2001** [comment]
> Hey parul , every things looks good but i could suggest you one thing, When your filtering categories instead of sending payload for each category, you can think of a logic where you to have send a single payload for filtering categories.So tommorow if your no of category increases in your backend you dont have to write separate code to handle it.

thanks for your suggestion , I will definately try to do this.

**Comment by parulgupta2001** [comment]
Thanks for your suggestion, I will definately try this.

**Review by srinibas39 — APPROVED** [review]

File: my-app/src/frontend/contexts/filter-context.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      console.log(response.data.products);
+    });
+  }, []);
+
+  const sortedData = getSortedData(productList, state.sortBy);
+  const rangeData = getRangeData(sortedData, state.rangeAll);
+  const stockData = getStockData(rangeData, state.inventoryAll);
+  const deliveryData = getDeliveryData(stockData, state.deliveryAll);
+  const ratingData = getRatingData(deliveryData, state.ratingFilter);
+  const categoryData = getCategoryData(ratingData, state.categoryAll);
```
Comment (iRohitGaur) [inline my-app/src/frontend/contexts/filter-context.js:45]: appreciation: this is a nice and clean approach

**Review by iRohitGaur — APPROVED** [review]
Suggestion: Please provide a preview link or screenshots of UI to get a better idea.

Nice work! Good to merge 👍🏻

**Pull request merged**